service/bookturks/quiz/QuizTools.py

```diff
@@ -238,3 +239,21 @@ def unlink_all_tags_of_quiz(quiz):
         """
         for tag in quiz.quiztag_set.all():
             QuizTagAdapter.unlink_quiz(tag_name=tag.tag_name, quiz_id=quiz.quiz_id)
+
+    @staticmethod
+    def check_attempt_eligibility(self, request, quiz_complete_model, quiz_id):
+        """
+        Checks if the user still has number of attempts left for this quiz
+        :return:
+        """
+        attempted = 0
+        user_profile_model = request.session.get(USER_PROFILE_MODEL)
+        attempted_quiz = user_profile_model.attempted_quiz
+        for quiz_result in attempted_quiz:
+            if quiz_result.quiz_model.quiz_id == quiz_id:
+                attempted = quiz_result.attempts
+                break
+
+        valid_attempts = int(quiz_complete_model.attempts)
+        if attempted >= valid_attempts:
+            return 1
```

Comment:
Like the other functions, I would suggest we can raise an exception rather than returning 1 with the appropriate message.
In this case "Attempts exceeded"
And handle this in the try catch in the view.